spec/controllers/starburst/announcements_controller_spec.rb

```diff
@@ -2,30 +2,41 @@
 
 module Starburst
 
-	describe AnnouncementsController do
-		
+	describe AnnouncementsController, type: :controller do
+
 		routes { Starburst::Engine.routes } # http://pivotallabs.com/writing-rails-engine-rspec-controller-tests/
 
 		it "marks an announcement as read (twice)" do
-			@current_user = mock_model(User, :id => 10)
-			controller.stub(:current_user).and_return(@current_user)
+			current_user = instance_double(User, id: 10)
+			controller.stub(:current_user).and_return(current_user)
 			announcement = FactoryGirl.create(:announcement)
 			announcement2 = FactoryGirl.create(:announcement)
-			post :mark_as_read, :id => announcement.id
+			if Rails::VERSION::MAJOR < 5
```

Comment:
I've had to add some special handling to those tests since the test helpers [introduced a breaking change after Rails 5.0](https://relishapp.com/rspec/rspec-rails/v/3-8/docs/controller-specs/controller-spec#setting-a-different-content-type-for-example-json-(request-type)).